### PR TITLE
Changed endpoint format for GET,PUT, DELETE for multiple objects of Class and Collections

### LINKED ISSUE #587: Change GET PUT POST DELETE requests format for collections
### I'm submitting a 
[x] bug report.

### Current Behaviour:
As discussed in the confcall, the current API semantics for operations on collections is far from being conventional:
```
/serverapi/MovieCollection/4f61070a-f647-4a65-9b0a-cbab4bcf0510/delete/3e1462f8-93ad-4a52-a371-a6fd5724c904, 1f6f1df8-788d-430a-8398-2e3302407fc0
```
Two major points:
* there is the repetition of "delete" in the uri
* multiple ids are listed with commas in the path

### Expected Behaviour:
The correct semantics should be
```
DELETE /serverapi/MovieCollection/4f61070a-f647-4a65-9b0a-cbab4bcf0510/?instances=3e1462f8-93ad-4a52-a371-a6fd5724c904,1f6f1df8-788d-430a-8398-2e3302407fc0
```
The operation method should be only defined in the request method and the list of ids to delete should be defined in a proper `instances` parameter. Every HTTP method can accept URL parameters (including PUT and DELETE).

### PR DESCRIPTION
**Pull request by farazkhanfk7**
Fixes #587

### Checklist
- [x] My branch is up-to-date with upstream/develop branch.
- [x] Everything works and tested for Python 3.6.0 and above.

### Current behaviour

The current API's endpoint format for operations on collections and deletion/insertion of multiple objects is :
```
/serverapi/MovieCollection/4f61070a-f647-4a65-9b0a-cbab4bcf0510/delete/3e1462f8-93ad-4a52-a371-a6fd5724c904, 1f6f1df8-788d-430a-8398-2e3302407fc0
```
Which isn't conventional and there is repetition of keywords like `add` and `delete` .

**Also, we had lot of code repetition in resource.py ( Items, ItemMember, ItemMembers to operate on single and multiple objects ) when most of it can be done using Item and ItemCollection resource classes with some additional params in request endpoint.**

### New expected behaviour

Previously, if a user wanted to create multiple objects of a Class, he/she had to pass ids in endpoints as well which can be really time consuming if one wants to PUT multiple objects by making request to a Class endpoint.

## PUT (without ids):
```
/serverapi/Movie
```
Request Body: 
```
{
  "data": [
    {
      "@type": "Movie",
      "movie_name": "Avengers",
      "movie_director": "Joss Whedon"
    },
    {
      "@type": "Movie",
      "movie_name": "Justice League",
      "movie_director": "Zack Snyder"
    }
  ]
}
```

## PUT ( with ids ):
```
http://localhost:8080/serverapi/Movie?instances=id1,id2,id3
```
Response : 
```
{
    "@context": "https://raw.githubusercontent.com/HydraCG/Specifications/master/spec/latest/core/core.jsonld",
    "@type": "Status",
    "description": "Objects with ID ['98c5a86f-b324-4270-b39b-0af0b7f99d07', '99c5a86f-b324-4270-b39b-0af0b7f99d12'] successfully added",
    "iri": [
        "http://localhost:8080/serverapi/Movie/98c5a86f-b324-4270-b39b-0af0b7f99d07",
        "http://localhost:8080/serverapi/Movie/99c5a86f-b324-4270-b39b-0af0b7f99d12"
    ],
    "statusCode": 201,
    "title": "Objects successfully added"
}
```

## DELETE ( multiple Class objects):
```
http://localhost:8080/serverapi/Movie?instances=id1,id2,id3
```
Response : 
```
{
    "@context": "https://raw.githubusercontent.com/HydraCG/Specifications/master/spec/latest/core/core.jsonld",
    "@type": "Status",
    "description": "Objects with ID ['98c5a86f-b324-4270-b39b-0af0b7f99d07', '99c5a86f-b324-4270-b39b-0af0b7f99d12'] successfully deleted",
    "statusCode": 200,
    "title": "Objects successfully deleted"
}
```

## DELETE ( Multiple members from a Collection)
```
http://localhost:8080/serverapi/MovieCollection/<collection_id>?instances=id1,id2,id3
```
Response:
```
{
    "@context": "https://raw.githubusercontent.com/HydraCG/Specifications/master/spec/latest/core/core.jsonld",
    "@type": "Status",
    "description": "Objects with ID ['27a3272d-c83d-4b96-8494-2799ac5bd67e', 'ac231414-a1e6-42f2-b6f4-38ec8fbacc4d'] successfully deleted",
    "statusCode": 200,
    "title": "Objects successfully deleted"
}
```

### CODE REVIEW

**Comment by Mec-iS** [comment]
Please use the `?instances=` syntax only for operations on multiple objects to accept lists.
there is no reason to change the semantics for single-instance operations, stick to what has been done on the non-collections endpoints.

**Review by Mec-iS — DISMISSED** [review]
See my comment  https://github.com/HTTP-APIs/hydrus/pull/590#issuecomment-864394665

**Comment by farazkhanfk7** [comment]
Changed query parameter to `?instances=` for all endpoints.

File: hydrus/resources.py [hunk 1]
```diff
@@ -106,6 +106,13 @@ def get(self, id_: str, path: str) -> Response:
             # Get path of the collection-class
             class_path = item_class.path
             is_collection = True
+            # To check if it's a request to GET a single instance from Collection
+            params = request.args.to_dict()
+            member_id = params.get('instances', None)
+            if member_id:
```
Comment (Mec-iS) [inline hydrus/resources.py]: please be explicit with checks: `if member_id is not None`.
Also it is better to be consistent, below you correctly wrote `params.get('instances')`, why in here we have `...get('instances', None)`. `None` is a default so it should not be expressed.

Comment (farazkhanfk7) [reply hydrus/resources.py]: Sure. changes made

**Review by Mec-iS — APPROVED** [review]

**Pull request merged**